descwl_coadd/coadd.py

```diff
@@ -89,6 +94,136 @@ def make_coadd_obs(
 def make_coadd(
     exps, coadd_wcs, coadd_bbox, psf_dims, rng, remove_poisson,
     max_maskfrac=MAX_MASKFRAC,
+    is_warps=False,
+):
+    """
+    make a coadd from the input exposures, working in "online mode",
+    adding each exposure separately.  This saves memory when
+    the exposures are being read from disk
+
+    Parameters
+    ----------
+    exps: list
+        Either a list of exposures or a list of DeferredDatasetHandle
+    coadd_wcs: DM wcs object
+        The target wcs
+    coadd_bbox: geom.Box2I
+        The bounding box for the coadd within larger wcs system
+    psf_dims: tuple
+        The dimensions of the psf
+    rng: np.random.RandomState
+        The random number generator for making noise images
+    remove_poisson: bool
+        If True, remove the poisson noise from the variance
+        estimate.
+    max_maskfrac: float
+        Maximum allowed masked fraction.  Images masked more than
+        this will not be included in the coadd.  Must be in range
+        [0, 1]
+    is_warps: bool
+        If set to True the input exps are actually handles for a data set, from
+        which the warps and info can be loaded
+    Returns
+    -------
+    coadd_data : dict
+        A dict with keys and values:
+
+            nkept: int
+                Number of exposures deemed valid for coadding
+            coadd_exp : ExposureF
+                The coadded image.
+            coadd_noise_exp : ExposureF
+                The coadded noise image.
+            coadd_psf_exp : ExposureF
+                The coadded PSF image.
+            coadd_mfrac_exp : ExposureF
+                The fraction of SE images interpolated in each coadd pixel.
+    """
+
+    check_max_maskfrac(max_maskfrac)
+
+    filter_label = exps[0].getFilter()
+
+    # this is the requested coadd psf dims
+    check_psf_dims(psf_dims)
```

Comment:
This should happen at the config time in the task framework. `check_psf_sims` should return `True` if valid and `False` if not.